lib/segment/src/index/field_index/full_text_index/posting_list.rs

```diff
@@ -0,0 +1,162 @@
+use bitvec::prelude::*;
+
+use crate::types::PointOffsetType;
+
+#[derive(Clone, Debug)]
+pub enum PostingList {
+    FullVec(Vec<PointOffsetType>),
+    BitVec(BitVec),
+}
+
+impl Default for PostingList {
+    fn default() -> Self {
+        Self::FullVec(vec![])
+    }
+}
+
+impl PostingList {
+    const LOWER_THRESHOLD: f64 = 0.03;
+    const UPPER_THRESHOLD: f64 = 0.032;
```

Comment:
this looks arbitrary, is there any reasoning why those values should be like this?

Reply:
I thought since `PointOffsetType=u32` so the flipping point in memory storage happens we will need to store `n` documents such that `n * 32 bit = N` where N is the total number of documents since each document takes only one bit in bitvec. so the ideal threshold should be `1/32≈0.3125`. In order to prevent back and forth conversion I expanded the range a bit to those values.